### ISSUE TITLE
Words with hyphen behaving erratically while implementing suggestions

### ISSUE THREAD

**Issue by jonptr** [florisboard/florisboard#2225]
#### Short description
I tried adding suggestions into florisboard. I'm now getting suggestions for the words now. But I'm facing an issue. For example I'm trying to type in 'check-in'. I type till 'check-' and my suggestions show 'check-in', if I click on the suggested 'check-in', instead of replacing 'check-' with 'check-in', it replaces it as 'check-check-in'. I think its becuase the last word is being identified wrongly and doesn't include the check word in it. I'm trying to figure out the code but I'm unable to do so. Could you help which function I need to debug and fix so I can correct this behaviour?

#### Steps to reproduce
1.Add some suggestions which has hyphen
2. Type in a word till the hyphen and wait for its suggestion to show up. for example type till 'check-' and suggestions 'check-in', 'check-out' etc will show up. 
3. Click on one of the suggestions.
4. Expected output, the word should get replaced as 'check-in'
5. Actual output, the word is getting replaced as 'check-check-in'

#### Environment information
- FlorisBoard Version: Latest version
- Install Source: Github
- Device: Any device

**Comment by jonptr** [comment]
@patrickgold Can you please help me with this issue?

**Comment by patrickgold** [comment]
That's true, hyphens are currently a big pain issue. However please note that anything regarding NLP is currently a hot zone in terms of changes, as I am working in #2209 on integrating and using my NLP implementation from [florisboard/nlp](https://github.com/florisboard/nlp). As such I'd suggest you wait with your work until I merge that in, especially as I will completely change the LatinProvider code and also some specific parts of the EditorInstance, which is responsible for determining what's a word and what's not.

**Comment by jonptr** [comment]
Hi @patrickgold ,

Thanks for the response. I'm trying to implement a third party NLP library and it works pretty well for the languages I'm targeting like French. However, since French has a lot of compound nouns with hyphens and this issue is causing a lot of concerns. I'm trying to debug and make changes to the code on how to determine what's a word or not. So could you please give some help on what functions I need to look at and modify, so I can try doing this myself?

**Comment by jonptr** [comment]
Hi @patrickgold I'm facing a lot of issues due to the wordsplit logic when integrating my NLP into the keyboard. Could you please provide some help in figuring out the word split logic so I can go and try to change it to suit my needs. Hope you understand.
